[Emmanuel2001] iP

Emmanuel2001 opened the pull request:
# BuddyTalk 
### Your Personal Task Assistant

>“Success is not final, failure is not fatal: it is the courage to continue that counts.” ― Winston S. Churchill ([source](https://www.goodreads.com/quotes/3270-success-is-not-final-failure-is-not-fatal-it-is))

Here's what BuddyTalk can do, and make your task management easier:
- Keeps track of your basic tasks (todo), deadlines, and events
- Easy search up for tasks using Find keyword
- Easy addition and deletion of tasks
- Marking and Unmarking tasks
- More coming soon!

Why BuddyTalk?
1. Runs Locally
2. Your own Personal Assistant 👍
3. **FREE**!! 

Features:
- [x] Managing tasks :books:
- [x] Managing deadlines
- [x] Find function

---

I used Java to build the app. Here is the `main` entry point of the application.
```java
public static void main(String[] args) {
        new BuddyTalk().run();
    }
```

`Try BuddyTalk Today!`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Emmanuel2001 committed:
Level-0

Emmanuel2001 committed:
Level-0

Emmanuel2001 committed:
Level-0

Emmanuel2001 committed:
Level-1

Emmanuel2001 committed:
Level-2

Emmanuel2001 committed:
Level-3

Emmanuel2001 committed:
Level-4

Emmanuel2001 committed:
Level-5

Emmanuel2001 committed:
Level-6

Emmanuel2001 committed:
A-TextUiTesting

Emmanuel2001 committed:
Made changes to the UI and added A-Enums

Emmanuel2001 committed:
Added storage and load data on run

Emmanuel2001 committed:
Merge branch 'branch-Level-7', added storage and load data on run

Emmanuel2001 committed:
Created branch level 8

Emmanuel2001 committed:
Add date and time for level 8

Emmanuel2001 committed:
Merge branch 'branch-Level-8', add date and time

Emmanuel2001 committed:
Add more OOP with TaskList, Parser and Command

Emmanuel2001 committed:
Add divide classes into packages

Emmanuel2001 committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Emmanuel2001 committed:
Add JUnit tests on deadline, event and taskList

Emmanuel2001 committed:
Add BuddyTalk as JAR file

Emmanuel2001 committed:
Add JavaDoc comments onto all files

Emmanuel2001 committed:
Tweak code to comply with coding standard

Emmanuel2001 committed:
Add finding of task with keyword

Emmanuel2001 committed:
Merge remote-tracking branch 'origin/branch-A-JavaDoc'

Emmanuel2001 committed:
Merge remote-tracking branch 'origin/branch-A-CodingStandard'

Emmanuel2001 committed:
Merge remote-tracking branch 'origin/branch-Level-9'

# Conflicts:
#	src/main/java/buddytalk/ui/BuddyTalk.java
#	src/main/java/buddytalk/ui/TaskList.java

nat-ong555 commented on src/main/java/buddytalk/ui/TaskList.java:
```diff
@@ -0,0 +1,87 @@
+     *
+     * @return An {@code ArrayList} of all the {@link Task} objects in the list.
+     */
+    public ArrayList<Task> getAllTasks() {
```
I like how the methods' names follow the naming convention and describes what it does. Good job!

nat-ong555 commented on src/main/java/buddytalk/ui/Event.java:
```diff
@@ -0,0 +1,80 @@
+    private static final DateTimeFormatter INPUT_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+
+    /** Formatter for displaying date and time, using "MMM dd yyyy, h:mm a" in US locale. */
+    private static final DateTimeFormatter DISPLAY_FORMAT = DateTimeFormatter.ofPattern("MMM dd yyyy, h:mm a", Locale.US);
```
The naming follows Java Coding Standard. Good job!

nat-ong555 commented on src/main/java/buddytalk/ui/Storage.java:
```diff
@@ -0,0 +1,99 @@
+        boolean isDone = Integer.parseInt(parts[1]) == 1;
+
+        return switch (taskType) {
+            case "T" -> new ToDo(parts[2], isDone);
```
The code block for the switch cases looks neat with correct indentation, nice work!

nat-ong555 commented on src/main/java/buddytalk/ui/ToDo.java:
```diff
@@ -0,0 +1,44 @@
+package buddytalk.ui;
+
+/**
+ * Represents a "ToDo" task. A {@code ToDo} is a basic task with a description and completion status,
+ * but no additional time or date information.
+ * <p>
+ * This class extends {@link Task} and implements specific formatting for representation
+ * and file storage.
+ * </p>
+ */
```
I like how you have header comments for each public class!

Ditzchann commented on src/main/java/buddytalk/ui/BuddyTalk.java:
```diff
@@ -0,0 +1,184 @@
+                    } catch (IllegalAccessException e) {
+                        Ui.displayError("Description for ToDo cannot be empty.");
+                    }
+                }
```
Why are different types of errors thrown for similar errors for different sub-classes of Task? Deadline throws an Exception for invalid input and Event throws a DateTimeParseException while ToDo throws an IllegalAccessException. Could they throw the same error instead for invalid/empty input?

Ditzchann commented on src/main/java/buddytalk/ui/Ui.java:
```diff
@@ -0,0 +1,114 @@
+                """;
+        start = start.indent(4);
+        System.out.println(start);
+    }
```
Could consider the use of \n and concatenating strings instead of using """?

Ditzchann commented on src/main/java/buddytalk/ui/TaskList.java:
```diff
@@ -0,0 +1,87 @@
+     */
+    public int size() {
+        return tasks.size();
+    }
```
Method could be called getSize() instead to represent an action?

Ditzchann commented on src/main/java/buddytalk/ui/ToDo.java:
```diff
@@ -0,0 +1,44 @@
+     */
+    public ToDo(String taskDescription, boolean isDone) {
+        super(taskDescription, TaskType.TODO, isDone);
+    }
```
Detailed javadoc comments for each method, good job!

Ditzchann commented on src/main/java/buddytalk/ui/Ui.java:
```diff
@@ -0,0 +1,114 @@
+                """, message);
+        str = str.indent(4);
+        System.out.println(str);
+    }
```
Some printed statements earlier was named text instead of str, do you want to rename it for consistency?

Ditzchann reviewed (commented):
LGTM! Just a couple of nits here and there but all's good!

Emmanuel2001 committed:
Add checkstyle and align code with coding standards

Emmanuel2001 committed:
Merge remote-tracking branch 'origin/branch-A-CheckStyle'

Emmanuel2001 committed:
Add GUI

Emmanuel2001 committed:
Merge remote-tracking branch 'origin/branch-Level-10'

Emmanuel2001 committed:
Modify code to adhere to code quality standard

To improve maintainability and readability of the code

Ensure code follows code quality such as SLAP

Allow readers to read the code at ease

Emmanuel2001 committed:
Add assert to code

Emmanuel2001 committed:
Add code quality to parser class

This helps maintain code readability and maintainability

Allow readers to have easier time to understand code

Emmanuel2001 committed:
Merge pull request #2 from Emmanuel2001/branch-A-Assertions

Add assert to code

Emmanuel2001 committed:
Merge pull request #3 from Emmanuel2001/branch-A-CodeQuality

Add code quality to parser class

Emmanuel2001 committed:
Merge branch 'master' of https://github.com/Emmanuel2001/ip

Emmanuel2001 committed:
Add help extension

Emmanuel2001 committed:
Add better gui and personality to buddytalk

Emmanuel2001 committed:
Add more tests

Emmanuel2001 committed:
Add Ui.png in docs

Emmanuel2001 committed:
Update README.md

Emmanuel2001 committed:
Merge branch 'master' of https://github.com/Emmanuel2001/ip

Emmanuel2001 committed:
Update README.md

Emmanuel2001 committed:
Update readme

Emmanuel2001 committed:
Revert "Update readme"

This reverts commit 1a8b2660831f043eebc30279bc2215cf7f86feb1.

Emmanuel2001 committed:
Update readme

Emmanuel2001 committed:
Update README.md

Emmanuel2001 committed:
Merge branch 'master' of https://github.com/Emmanuel2001/ip

Emmanuel2001 committed:
Add link to release on readme

Emmanuel2001 committed:
Fix errors

Emmanuel2001 committed:
Update README.md

Emmanuel2001 committed:
Fix Error

Emmanuel2001 committed:
Update README.md

Emmanuel2001 committed:
Add fixes and update readme

Emmanuel2001 committed:
Update readme

Emmanuel2001 committed:
Update README.md

Emmanuel2001 committed:
Update release notes

Emmanuel2001 committed:
Update README.md
